### ISSUE TITLE
VS Code specific environment variables shouldn't be passed on to debug targets

### ISSUE THREAD

**Issue by toptensoftware** [microsoft/vscode#100166]
Issue Type: <b>Bug</b>

1. Under Linux, start debugging any dotnet project that can log out the environment variables.
2. Notice the debug target's environment variables include many related to VS Code's configuration.

For example, running any debug target that requires Wayland on Linux will fail because Wayland is disabled by VS Code's configuration. ie: the DISABLE_WAYLAND and changes to the XDG_* (especially XDG_RUNTIME_DIR) variables should be undone before launching the debug target.

VS Code version: Code 1.46.0 (a5d1cc28bb5da32ec67e86cc50f84c67cc690321, 2020-06-10T08:59:04.923Z)
OS version: Linux x64 5.3.0-59-generic snap

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-3930K CPU @ 3.20GHz (12 x 1831)|
|GPU Status|2d_canvas: unavailable_software<br>flash_3d: disabled_software<br>flash_stage3d: disabled_software<br>flash_stage3d_baseline: disabled_software<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: disabled_off<br>rasterization: disabled_software<br>skia_renderer: disabled_off_ok<br>video_decode: disabled_software<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: unavailable_software<br>webgl2: unavailable_software|
|Load (avg)|1, 1, 1|
|Memory (System)|46.99GB (41.10GB free)|
|Process Argv|. --no-sandbox|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu-wayland|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu-wayland|
|XDG_SESSION_TYPE|wayland|
</details><details><summary>Extensions (3)</summary>

Extension|Author (truncated)|Version
---|---|---
docomment|k--|0.1.11
csharp|ms-|1.22.0
cpptools|ms-|0.28.3

</details>

**Comment by isidorn** [comment]
Forwarding first to @weinand since he looked into environment variables passing. And then he can potentialy forward to dotnet extension

**Comment by weinand** [comment]
This is a valid request.

Until we'll have a fix, please try this:

Some debug extensions not only support to specify environment variables for the debug target, but also to delete environment variables by setting them to `null`, e.g.
```js 
"env": {
   "foo": "bar",
   "DISABLE_WAYLAND": null
}
```
Please try whether this works for your debug extension (or file a feature request against the extension).

**Comment by toptensoftware** [comment]
Thanks @weinand,

Yep I already got it working, but it took half and hour of trial and error to figure out exactly which variable was causing the problem - which is why I logged this issue. 

Similarly, I'd noticed previously when doing some GTK+ development that that app behaved differently when run from VSCode compared to running from command line.  It didn't occur to me at the time that it'd be an environment variable - but I'm sure that's what it was.

Basically, if this could be fixed it'd save others from the seeing mysterious differences between running from VSCode and launching from command line.

Brad

**Comment by connor4312** [comment]
Duplicates https://github.com/microsoft/vscode/issues/42373
